sdk/python/kfp/components/component_factory_test.py

```diff
@@ -45,6 +47,15 @@ def test_with_packages_to_install_with_pip_index_url(self):
         for package in packages_to_install + pip_index_urls:
             self.assertTrue(package in concat_command)
 
+    def test_output_named_Output(self):
+
+        with self.assertRaisesRegex(ValueError,
+                                    r'"Output" is an invalid parameter name.'):
+
+            @component
+            def comp(Output: OutputPath(str), text: str):
+                return text
```

Comment:
nit:
Even though this component will never be used, I think it makes sense to either make it a valid component or a component with no body at all (`...`/`pass`). Otherwise the test is suggestive of invalid usage.

The interface of this function declares one input of type `str` and one output of type `str`, outputted via `OutputPath`. While return `return text` does output a string, it does not do so via the `OutputPath` style. See [this example](https://github.com/kubeflow/pipelines/blob/4cc0e801830cc20efd59389b8b286ff31d43398b/sdk/python/test_data/components/preprocess.py#L38) for outputting parameters via `OutputPath`.